### DISCUSSION TITLE
[Enhancement] 页面在使用zoom或scale方案放大缩小后，悬浮框错位

### DISCUSSION THREAD

**Discussion by Ciaxinwen** [element-plus/element-plus#19266] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `All`

## Description

页面在使用zoom或scale方案放大缩小后，基于PopperJs封装的悬浮框组件均偏移，包括tooltip、popover、dropdown等，望适配

### Additional comments

_(empty)_

**Comment by Dsaquel** [comment]
The element where you set the zoom you need to set `position: relative;`
[issue related](https://github.com/element-plus/element-plus/issues/18970)
[Playground example](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICAgICAgPGVsLXRvb2x0aXBcbiAgICAgICAgY2xhc3M9XCJib3gtaXRlbVwiXG4gICAgICAgIGVmZmVjdD1cImRhcmtcIlxuICAgICAgICBjb250ZW50PVwiTGVmdCBCb3R0b20gcHJvbXB0cyBpbmZvXCJcbiAgICAgICAgcGxhY2VtZW50PVwibGVmdC1lbmRcIlxuICAgICAgPlxuICAgICAgICA8ZWwtYnV0dG9uPnRvcC1lbmQ8L2VsLWJ1dHRvbj5cbiAgICAgIDwvZWwtdG9vbHRpcD5cbiAgICA8ZWwtZHJvcGRvd24gcGxhY2VtZW50PVwiYm90dG9tLWVuZFwiPlxuICAgICAgPGVsLWJ1dHRvbj4gYm90dG9tRW5kIDwvZWwtYnV0dG9uPlxuICAgICAgPHRlbXBsYXRlICNkcm9wZG93bj5cbiAgICAgICAgPGVsLWRyb3Bkb3duLW1lbnU+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAxc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAyc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAzc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgIDwvZWwtZHJvcGRvd24tbWVudT5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgPC9lbC1kcm9wZG93bj5cbjwvdGVtcGxhdGU+XG5cbjxzdHlsZT5cbmJvZHl7XG4gIHpvb206IDAuNztcbiAgcG9zaXRpb246IHJlbGF0aXZlOyAvKiB0b2dnbGUgdGhpcyBsaW5lICovXG59XG48L3N0eWxlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnt9fQ==)
Keep in mind there are some components behavior that might not work well with `zoom` ([ref](https://github.com/element-plus/element-plus/issues/18994)), use it with caution.

**Comment by zjpzjp** [comment]
element-plus大部分弹出都基于popper.js 
给body增加position:relative 即可
